file_store/src/file_store.rs

```diff
@@ -57,27 +57,27 @@ impl FileStore {
         })
     }
 
-    pub async fn list_all<A, B, F>(
+    pub async fn list_all<A, B, P>(
         &self,
-        file_type: F,
+        file_type: P,
         after: A,
         before: B,
     ) -> Result<Vec<FileInfo>>
     where
-        F: Into<FileType> + Copy,
+        P: ToString + std::fmt::Display,
```

Comment:
I believe this Display bound is unnecessary.

Generally, I would not suggest having things impl ToString, and instead just have the function take a `&str`

Reply:
addressed; i thought we had steered toward a slightly more permissive `impl ToString` and `impl AsRef<str>` but maybe i'm thinking of something else